js/tabState.js

```diff
@@ -130,9 +131,17 @@ var tasks = {
       task = {}
     }
 
+    var tabStack
+    if (task.history) {
+      tabStack = Object.setPrototypeOf(task.history, TabStack.prototype)
```

Comment:
I think it would be better if the tabStack constructor could take an object to set the stack to. This could look like this:

```
new TabStack(task.history)
```